create_venvs.py

```diff
@@ -55,19 +56,45 @@ def create_venv(path: Path, recreate: bool = False):
     pip_install(venv_path, "-U", "pip")
     print("")
 
-    requirements_path = str(path / "requirements.txt")
+    requirements_path = generate_requirements(path)
     print(f"> Install dependencies from {requirements_path}...")
-    pip_install(venv_path, "-r", requirements_path)
+    pip_install(venv_path, "-r", str(requirements_path))
     print("")
 
-    print("> Install/Upgrade development tools: pydocstyle, pyright, ruff...")
-    pip_install(venv_path, "-U", "pydocstyle", "pyright", "ruff")
+    dev_tools = ["pydocstyle", "pyright", "ruff"]
+    print(f"> Install/Upgrade development tools: {dev_tools}...")
+    pip_install(venv_path, "-U", *dev_tools)
     print("")
 
     print(f">>> Finished creating a virtual environment under the path {venv_path}.")
     print("")
     print("")
 
+def generate_requirements(path: Path) -> Path:
+    """
+    If the requirements.txt file at the path needs to be updated for local developement, generate
+    a new requirements file.
+
+    Return the path to the requirements file to be used.
+
+    :param path: The path to the directory containing the requirements.txt file.
+    """
+    requirements_path = path.joinpath("requirements.txt")
+
+    if not re.search(r"images\/airflow\/[2-3]\.[0-9]+\.[0-9]+$", str(path.resolve())):
```

Comment:
Not sure I follow the logic here. If the requirements.txt file is not found, then we treat it as the development setup. Is that correct? And if so, I don't understand why that would be true for only a development setup.

Reply:
We have multiple requirements.txt in this project, only the one in the path checked above needs to be updated for a development build.

The mistake I've made is that I thought this script was used only for setting up a development environment, but it looks like we do use it as part of the buildspec for the image generation.

I'll have to rethink this as an entrypoint, maybe looking to add a parameter to indicate a development/production build.